### PR TITLE
BUGFIX: RAIL-1161 AFMTable component uses empty itemIdentifiers array in resultSpec measure dimension when AFM doesn't contain measures

### PR DESCRIPTION
**Pull request by ocetnik**
`6.0.0-ocetnik-doc-table-without-measure-next-2018-10-12T06-23-20-555Z`

### CODE REVIEW

**Comment by ocetnik** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by ocetnik** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1738/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1218/ : SUCCESS in 11m 30s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/716/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1739/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/381/ : SUCCESS in 24m 25s

**Comment by ocetnik** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-597.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1740/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/382/ : FAILURE in 28m 38s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1741/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/383/ : SUCCESS in 10m 34s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1746/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1224/ : SUCCESS in 6m 23s

**Comment by ocetnik** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/718/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1747/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/384/ : SUCCESS in 17m 25s

**Comment by ocetnik** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-597.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1748/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/385/ : SUCCESS in 23m 24s

File: src/components/afm/Table.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ function generateDefaultDimensions(afm: AFM.IAfm): AFM.IDimension[] {
             itemIdentifiers: (afm.attributes || []).map(a => a.localIdentifier)
         },
         {
-            itemIdentifiers: ['measureGroup']
+            itemIdentifiers: (afm.measures) ? ['measureGroup'] : []
```
Comment (tomasreichmann) [inline src/components/afm/Table.tsx]: no need for ()

File: src/components/afm/tests/Table.spec.tsx [hunk 2]
```diff
@@ -5,10 +5,10 @@ import { testUtils } from '@gooddata/js-utils';
 import { Table } from '../Table';
 import { SortableTable } from '../../core/SortableTable';
 import { dummyExecuteAfmAdapterFactoryTable } from './utils/DummyExecuteAfmAdapter';
-import { executionRequest } from './utils/dummyFixture';
+import { executionRequest, executionRequestWithoutMeasureAndWithoutResultSpec } from './utils/dummyFixture';
 
 describe('Table', () => {
-    it('should provide default resultSpec to the core Table with attributes', () => {
+    it('should provide default resultSpec to the core Table with attribute and measure', () => {
         const wrapper = mount((
```
Comment (tomasreichmann) [inline src/components/afm/tests/Table.spec.tsx]: extra ()

File: src/components/afm/tests/Table.spec.tsx [hunk 3]
```diff
@@ -27,4 +27,24 @@ describe('Table', () => {
             ]);
         });
     });
+
+    it('should provide default resultSpec to the core Table with no measure (with only attributes)', () => {
+        const wrapper = mount((
```
Comment (tomasreichmann) [inline src/components/afm/tests/Table.spec.tsx]: extra ()

**Comment by tomasreichmann** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1753/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1232/ : SUCCESS in 9m 04s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/408/ : SUCCESS in 32s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1234/ : SUCCESS in 10m 59s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/723/ : SUCCESS in 15m 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/330/ : SUCCESS in 21m 35s

**Pull request merged**